tests/cases/unittests/transpile.ts

```diff
@@ -0,0 +1,59 @@
+﻿/// <reference path="..\..\..\src\harness\harness.ts" />
+
+module ts {
+    describe("Transpile", () => {
+
+        function runTest(input: string, compilerOptions: ts.CompilerOptions = {}, expectedOutput?: string, expectedDiagnosticCodes: number[] = []): void {
+            let diagnostics: Diagnostic[] = [];
+            let result = transpile(input, compilerOptions, "file.ts", diagnostics);
+
+            for (let i = 0; i < expectedDiagnosticCodes.length; i++) {
+                assert.equal(expectedDiagnosticCodes[i], diagnostics[i] && diagnostics[i].code, `Could not find expeced diagnostic.`);

```

Comment:
I'd rather you checked the sizes instead of `diagnostics[i] &&`, and that you always print out the entire results if something is wrong.